### PR TITLE
Update Releases Notes for ONVM v19.07

### PR DESCRIPTION
**Pull request by koolzz**
Still working on it, I'll ping people when its ready for review.

### CODE REVIEW

**Comment by onvm** [comment]
> In response to PR creation

## CI Message

Your results will arrive shortly

**Comment by koolzz** [comment]
@onvm obrigado

**Comment by onvm** [comment]
> @onvm obrigado

## CI Message

Another CI run in progress, adding request to the end of the list

**Comment by onvm** [comment]
> @onvm obrigado

## CI Message

Your results will arrive shortly

**Comment by koolzz** [comment]
Feel free to make me feel bad about my grammar mistakes @twood02 @kevindweb @dennisafa

File: docs/Releases.md [hunk 1]
```diff
@@ -13,6 +13,41 @@ use a date based versioning system.  Now, a release version can look
 like `17.11` where the "major" number is the year and the "minor" number
 is the month.
 
+## v19.07 (7/19): NFD library and example NFs, minor improvements and bug fixes.
+A CloudLab template is available with the latest release here: https://www.cloudlab.us/p/GWCloudLab/onvm
+
+**Performance**: This release includes a new macro `ENABLE_FLOW_LOOKUP` which will disable an expensive flow lookup for every incoming packet(instead packets are forwarded to the default service ID). This improves performance but flow lookup is still enabled by default for backward compatibility with other applications that use ONVM.
```
Comment (twood02) [inline docs/Releases.md]:
```suggestion
**Performance**: This release includes a new macro `ENABLE_FLOW_LOOKUP` which controls whether a flow lookup is performed for every incoming packet. If disabled, all packets are forwarded to the default service ID which improves performance. The flow lookup is still enabled by default for backward compatibility with other applications that use ONVM.
```

File: docs/Releases.md [hunk 2]
```diff
@@ -13,6 +13,41 @@ use a date based versioning system.  Now, a release version can look
...
+
+### NFD library with example NFS
+Add NFD, a C++-based NF developing compiler designed by Wenfei's group (http://wenfei-wu.github.io/) from IIIS, Tsinghua University, China.
```
Comment (twood02) [inline docs/Releases.md]:
```suggestion
Add example NFs based on NFD, a C++-based NF developing compiler designed by Wenfei Wu's group (http://wenfei-wu.github.io/) from IIIS, Tsinghua University, China.
```

Comment (twood02) [reply docs/Releases.md]: Does the code we merged actually include the NFD compiler? Or is it just some NFs that were produced by the NFD tool?  I think it is the latter, so we will need to adjust the text.

File: docs/Releases.md [hunk 3]
```diff
@@ -13,6 +13,41 @@ use a date based versioning system.  Now, a release version can look
...
+ - DNS Amplification Mitigation
+ - Super Spread Detection
+ - Heavy Hitter Detection
+ - SYN Flood Detection
+ - UDP Flood Detection
+ - Stateless Firewall
+ - Stateful Firewall
+ - NAPT
+
+### CI updates:
```
Comment (twood02) [inline docs/Releases.md]:
```suggestion
### Continuous Integration updates:
```

File: docs/Releases.md [hunk 4]
```diff
@@ -13,6 +13,41 @@ use a date based versioning system.  Now, a release version can look
...
+CI got a few major updates this release:
+ - CI will do basic lint checks and branch checks(all PRs should be submitted against the **develop** branch) for unathorized users
+ - If CI is working on a request and recives another request it will append it to the queue instead of dropping it
+ - CI will now run Pktgen as an additional test metric.
+
+### Minor improvements
+ - Shared core functionality for messages
+ - NF core reassignment on termination
```
Comment (twood02) [inline docs/Releases.md]: Is reassignment enabled by default? Do we explain it in more detail anywhere else?  I suggest calling this "NF core reballancing"

File: docs/Releases.md [hunk 5]
```diff
@@ -13,6 +13,41 @@ use a date based versioning system.  Now, a release version can look
...
+ - NAPT
+
+### CI updates:
+CI got a few major updates this release:
+ - CI will do basic lint checks and branch checks(all PRs should be submitted against the **develop** branch) for unathorized users
+ - If CI is working on a request and recives another request it will append it to the queue instead of dropping it
+ - CI will now run Pktgen as an additional test metric.
+
+### Minor improvements
+ - Shared core functionality for messages
```
Comment (twood02) [inline docs/Releases.md]: What does this mean exactly? This has a new API, right? We should explain it.  We used to have a section in the release notes about new/changed APIs.

File: docs/Releases.md [hunk 6]
```diff
@@ -13,6 +13,41 @@ use a date based versioning system.  Now, a release version can look
...
+ - NF core reassignment on termination
+ - Fix Style guide links
```
Comment (twood02) [inline docs/Releases.md]: this sounds like a bug fix to me

File: docs/Releases.md (hunk 1 above)
Comment (twood02) [reply docs/Releases.md]: If disabled, does it forward to the default service ID or the default service *chain*?

File: docs/Releases.md [hunk 7]
```diff
@@ -13,6 +13,42 @@ use a date based versioning system.  Now, a release version can look
...
+ - Heavy Hitter Detection
+ - SYN Flood Detection
+ - UDP Flood Detection
+ - Stateless Firewall
+ - Stateful Firewall
+ - NAPT
+
+### Continuous Integration updates:
+CI got a few major updates this release:
+ - CI will do basic lint checks and branch checks(all PRs should be submitted against the *develop* branch) for unathorized users
```
Comment (kevindweb) [inline docs/Releases.md]: for `unauthorized` users

File: docs/Releases.md [hunk 8]
```diff
@@ -13,6 +13,42 @@ use a date based versioning system.  Now, a release version can look
...
+ - If CI is working on a request and recives another request it will append it to the queue instead of dropping it
```
Comment (kevindweb) [inline docs/Releases.md]: and `receives` another

File: docs/Releases.md [hunk 9]
```diff
@@ -13,6 +13,42 @@ use a date based versioning system.  Now, a release version can look
...
 
+## v19.07 (7/19): NFD library and example NFs, Continuous Integration updates, minor improvements and bug fixes.
+A CloudLab template is available with the latest release here: https://www.cloudlab.us/p/GWCloudLab/onvm
+
+**Performance**: This release includes a new macro `ENABLE_FLOW_LOOKUP` which controls whether a flow lookup is performed for every incoming packet. If disabled, all packets are forwarded to the default service ID which improves performance. The flow lookup is still enabled by default for backward compatibility with other applications that use ONVM.
+
+### NFD library with example NFS
+Add example NFs based on NFD, a C++-based NF developing compiler designed by Wenfei Wu's group (http://wenfei-wu.github.io/) from IIIS, Tsinghua University, China.
+
+NFD compiles the NF logic into common C++ program by using table-form language to model NFs' behavior . With NFD's help, NF developer, free from the problems that porting NFs into specific network environment or deploying platforms, can easily concentrate on designing NF's core logic.
```
Comment (dennisafa) [inline docs/Releases.md]: [nit] into a* common C++ program, extra space after behavior.

**Pull request merged**